:book: Add release templates (#9572)
kubernetes-sigs/cluster-api: docs/release/release-templates.md

> # Release Templates

Reviewer: Should we refer to this document from release tasks document, specifically from https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#continuously-communicate-key-dates-to-the-community?

Author: Sure, linked to this page here: https://github.com/kubernetes-sigs/cluster-api/pull/9572/commits/ebfcf48a381b42727964c4e2fd86729d228f1160#diff-751bae1afb6862debc4d8d6b2d39bba9d2838e720c37c8b2805138c738843025R381

This document contains a collection of announcement templates for the release team.

## Email Release Announcement Template

```
Hello everyone,

> Cluster API patch versions vX.Y.Z has been released!

Reviewer: ```suggestion
Cluster API patch version vX.Y.Z has been released!
```

This release was focused on improving existing features and stability of CAPI. 
There were a total of A commits and B bugs fixed by our awesome contributors! Kudos!

> Below are the release notes of each of the patch release :

Reviewer: ```suggestion
Below is the release note of the vX.Y.Z patch release:
```

- https://github.com/kubernetes-sigs/cluster-api/releases/tag/vX.Y.Z

Thanks to all our contributors!

Best, 
CAPI Release Team
```

## Slack Weekly Announcement Template

```

> # Weekly announcement template

Author: Yes although would it still be worth keeping around if someone wants to reference it? I don't see why anyone would need to though so I can go ahead and remove it.

> # Weekly announcement template

Reviewer: I'd probably leave it as a reference, since someone new in the team to the tool and never used it before to generate weekly announcements that will be introduced in #9343, has an idea of how it looks like beforehand.

> # Weekly announcement template

Reviewer 2: I just think it's going to be a lot of effort to keep it in sync vs just running the tool. Maybe the compromise is to have the templates for the notes tool be templates somewhere instead of being inlined in the go code?

> # Weekly announcement template

Author: I can add a line specifying that you can generate this template automatically with the notes tool. It might take some significant refactoring to have the generator to use it. Should I work on it now or defer it to a separate issue?

> # Weekly announcement template

Reviewer 2: Fine to keep it for now IMO if the team is happy with it - would be good to move to templating over time as it probably makes the whole process easier to change and adapt.
